src/delagent/agent/delagent.c

```diff
@@ -96,84 +140,124 @@ int main (int argc, char *argv[])
-      case 'c': GotArg=1; break; /* handled by fo_scheduler_connect() */
-      case 'V': printf("%s", BuildVersion); PQfinish(db_conn); return(0);
-      default:	Usage(argv[0]); exit(-1);
+      case 'f':
+        listFolder=1;
+        gotArg=1;
+        break;
+      case 'F':
+        delFolder=atol(optarg);
+        gotArg=1;
+        break;
+      case 'L':
+        delLicense=atol(optarg);
+        gotArg=1;
+        break;
+      case 's':
+        scheduler=1;
+        gotArg=1;
+        break;
+      case 'T':
+        Test++;
+        break;
+      case 'u':
+        listProj=1;
+        gotArg=1;
+        break;
+      case 'U':
+        delUpload=atol(optarg);
+        gotArg=1;
+        break;
+      case 'v':
+        Verbose++;
+        break;
+      case 'c':
+        gotArg=1;
+        break; /* handled by fo_scheduler_connect() */
+      case 'V':
+        printf("%s", BuildVersion);
+        PQfinish(db_conn);
+        return(0);
+      default:
+        usage(argv[0]);
+        exit(-1);
     }
   }
 
-  if (!GotArg)
-  {
-    Usage(argv[0]);
-    exit(-1);
-  }
-
-  if (Scheduler != 1 && 1 != authentication(user_name, password, &user_id, &user_perm)) 
+  if (!gotArg)
   {
-    LOG_FATAL("User name or password is invalid.\n");
+    usage(argv[0]);
     exit(-1);
   }
 
-  COMMIT_HASH = fo_sysconfig("delagent", "COMMIT_HASH");
-  VERSION = fo_sysconfig("delagent", "VERSION");
-  sprintf(agent_rev, "%s.%s", VERSION, COMMIT_HASH);
-  /* Get the Agent Key from the DB */
-  fo_GetAgentKey(db_conn, basename(argv[0]), 0, agent_rev, agent_desc);
-  
-  if (ListProj) ListUploads(user_id, user_perm);
-  if (ListFolder) ListFolders(user_id);
-
-  alarm(60);  /* from this point on, handle the alarm */
-  if (DelUpload) 
+  if (scheduler != 1)
   {
-    if (1 != check_permission_del(DelUpload, user_id, user_perm))
+    if (0 != authentication(user_name, password, &user_id, &user_perm))
     {
-      LOG_FATAL("You '%s' does not have the permsssion to delete the upload '%ld', or the upload '%ld' does not exist.\n", user_name, DelUpload, DelUpload);
+      LOG_FATAL("User name or password is invalid.\n");
       exit(-1);
     }
-    DeleteUpload(DelUpload); 
-    fprintf(stdout, "The upload '%ld' is deleted by the user '%s'.\n", DelUpload, user_name);
-  }
-  if (DelFolder) { DeleteFolder(DelFolder); }
-  if (DelLicense) { DeleteLicense(DelLicense); }
 
-  /* process from the scheduler */
-  if (Scheduler)
-  {
-    while(fo_scheduler_next())
+    COMMIT_HASH = fo_sysconfig("delagent", "COMMIT_HASH");
+    VERSION = fo_sysconfig("delagent", "VERSION");
+    sprintf(agent_rev, "%s.%s", VERSION, COMMIT_HASH);
+    /* Get the Agent Key from the DB */
+    fo_GetAgentKey(db_conn, basename(argv[0]), 0, agent_rev, agent_desc);
+
+    if (listProj)
     {
-      Parm = fo_scheduler_current();
-      
-      if (ReadParameter(Parm) < 0)
-        exit(-1);
+      returnedCode = listUploads(user_id, user_perm);
+    }
```

Comment:
if returnedCode !=  0 you might want to leave the program here before you set it again if listFolder is true